### PR TITLE
Update embedded CFamily analyzer to 6.48.0.62520

### LINKED ISSUE #4738: Update embedded CFamily analyzer to 6.48.0.62520
Update embedded CFamily analyzer to 6.48.0.62520

6.47.0.62356 → 6.48.0.62520

Release notes: https://sonarsource.atlassian.net/issues/?jql=project%20%3D%2010166%20AND%20fixVersion%3D14218%20ORDER%20BY%20priority%20DESC%2C%20key%20ASC

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #4738

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs [hunk 1]
```diff
@@ -44,9 +44,8 @@ public class CFamilyEmbeddedSonarWayRulesTests
         // The QP filter has "active/inactive" tabs. The number of rules is shown in the top-right of the screen.
         // 5. Repeat for C++.
 
-        // Rule data for C-Family plugin v6.47.0.62356
+        // Rule data for C-Family plugin v6.48.0.62520
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs:47]: Looking at the release notes, there don't appear to be any changes in the number of rules or the contents of Sonar Way.

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs [hunk 2]
```diff
@@ -44,14 +44,17 @@ public class CFamilyEmbeddedSonarWayRulesTests
...
+        // You can check the version of the plugin that is installed on the appropriate web API:
+        // e.g. https://next.sonarqube.com/sonarqube/api/plugins/installed and https://sonarcloud.io/api/plugins/installed
+        // Note - you need to be logged in.
+
+        // Rule data for C-Family plugin v6.48.0.62520
 
-        // Note: does not include MISRA rules - see #4604
         private const int Active_C_Rules = 210;
-        private const int Inactive_C_Rules = 123;
+        private const int Inactive_C_Rules = 122;
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs:54]: image

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs [hunk 3]
```diff
@@ -44,14 +44,17 @@ public class CFamilyEmbeddedSonarWayRulesTests
...
 
         private const int Active_CPP_Rules = 436;
-        private const int Inactive_CPP_Rules = 207;
+        private const int Inactive_CPP_Rules = 203;
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs:57]: image

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs [hunk 4]
```diff
@@ -78,19 +81,21 @@ public void Read_Active_Rules()
         [TestMethod]
         public void Read_Rules_Params()
         {
-            rulesMetadataCache.GetRulesConfiguration("cpp").RulesParameters.TryGetValue("S1311", out var parameters);
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs:81]: S1311 was marked as obsolete in an earlier version of SQ and removed in the latest release: see [CPP-4615](https://sonarsource.atlassian.net/browse/CPP-4615).

I just picked the first rule that had a `xxx_params.json` file.

**Comment by duncanp-sonar** [comment]
v6.48 of the CFamily analyzer has not yet been deployed to Next or SonarCloud, which makes checking the expected number of rules more difficult. However, it is do-able:

1. log in to Next/SonarCloud and check which version of the plugin is installed using the appropriate web API e.g. https://sonarcloud.io/api/plugins/installed
2. If it isn't the expected version, you can always install the plugin in a local copy of SonarQube (which is what I did). Just copy the Cfamily jar to the `lib\extensions` folder, replacing the old version (this assumes the plugin API hasn't changed between SQ versions):

image

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Comment by duncanp-sonar** [comment]
I'll cherry-pick the CFamily-related commits. The other commits are in the hardening branch.

**Pull request merged**